src/csharp/Grpc.Core/Internal/NativeExtension.cs

```diff
@@ -129,14 +129,11 @@ private static NativeMethods LoadNativeMethods()
         /// </summary>
         private static NativeMethods LoadNativeMethodsUnity()
         {
-            switch (PlatformApis.GetUnityRuntimePlatform())
-            {
-                case "IPhonePlayer":
-                    return new NativeMethods(new NativeMethods.DllImportsFromStaticLib());
-                default:
-                    // most other platforms load unity plugins as a shared library
-                    return new NativeMethods(new NativeMethods.DllImportsFromSharedLib());
-            }
+#if UNITY_IOS
```

Comment:
Given this .cs file is compiled outside of Unity (as a regular C# assembly), the UNITY_IOS will be always false
and the `return new NativeMethods(new NativeMethods.DllImportsFromStaticLib());` branch will never be hit.

Reply:
Hmm - I've actually been compiling this fix within Unity to avoid error. From my own debugging, https://github.com/grpc/grpc/issues/22485#issuecomment-604214839 causes the crash - are there any other way to get platform information?

Reply:
I guess one way to handle this is to rely on [RuntimePlatform](https://docs.unity3d.com/ScriptReference/RuntimePlatform.html), but I'm not sure if it's allowed to import UnityEngine?

Reply:
We use this code for platform detection https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Core/Internal/PlatformApis.cs,

but the codepath for detecting unity is know not to be ideal and there's been complaints about it in other issues.
But for now this is the best we have.